java/src/com/ibm/streamsx/topology/TStream.java

```diff
@@ -89,6 +90,41 @@
      * @return Filtered stream
      */
     TStream<T> filter(Predicate<T> filter);
+    
+    /**
+     * Split a stream into {@code n} streams as specified by {@code splitter}.
+     * 
+     * <P>
+     * For each tuple on the stream, {@code splitter.applyAsInt(tuple)} is called.
+     * The return value {@code r} determines the destination stream:
+     * <pre>
+     * if r < 0 the tuple is discarded
+     * else it is sent to the stream at position (r % n) in the returned array.
+     * </pre>
+     * </P>
+     * 
+     * <P>
+     * Example of round robin splitting a stream of {@code String} into 5 channels.
```

Comment:
related to that, shouldn't there also be a split(int n) that lacks a ToIntFunction, where, just like parallel, the behavior is: if the tuple supports Keyable<> that's used, otherwise it's round-robin.

also, shouldn't there be a parallel(..., ToIntFunction()), enabling arbitrary partitioning logic w/o the need to define new tuple types (and transforms) just to get different partitioning of the same logical tuple stream in different parallel regions?